### DISCUSSION TITLE
different HTML template content for production and dev

### DISCUSSION THREAD

**Discussion by ViliusS** [webpack/webpack#13021] [Q&A]
We have an external system where content of the HTML template is prefilled in production. For example if HTML template has %FOOTER_MACRO% it is prefilled with HTML content which is static and cannot be modified by us.

To prepare template for that system we have a Webpack project which emits one HTML file. I would like to some how define those % externally generated macros in a project, so it would generate full HTML page for development environment (hot server), but different "unfilled" % macros for production in /dist folder. i.e. under _npm run start:dev_ it would emit full HTML 

`<html><body>...<footer>blabla</footer></body></html>`

but under _npm run build_ it would produce just

`<html><body>...%FOOTER_MACRO%</body></html>`

Where would I define that %FOOTER_MACRO% = `<footer>blabla</footer>`?

We are using [static HTML webpack boilerplate](https://github.com/erickzhao/static-html-webpack-boilerplate) which uses WebPack4 I believe, if that makes a difference.

**Comment by alexander-akait** [comment]
Please ask this questions in `static-html-webpack-boilerplate` repo

**Reply by ViliusS** [reply]
It is a more general question. I'm asking what is the best way to do this? Maybe a combination of html-webpack-plugin and some other plugin I should know about? Or maybe there are native webpack features which could achieve this?

**Reply by alexander-akait** [reply]
You need plugin for this, we don't support expressions in HTML files
